docs/aws/audit/elbmonitoring/rules/elb_no_deletion_policy_remediation.mdx

```diff
@@ -1,6 +1,206 @@
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the EC2 Dashboard by selecting "Services" from the top menu, then choosing "EC2" under the "Compute" category.
+3. In the EC2 Dashboard, under the "Load Balancing" section in the left-hand navigation pane, click on "Load Balancers".
+4. In the Load Balancers page, select the load balancer you want to check. In the "Description" tab below, look for the "Attributes" section. Here, you can see if the "Deletion Protection" flag is enabled or not.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   ```
+   pip install awscli
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all the Load Balancers: Use the following command to list all the load balancers in your AWS account:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+   This command will return a JSON output with details of all the load balancers.
+
+3. Extract Load Balancer ARNs: From the JSON output, extract the ARN of each load balancer. You can use the following command to do this:
+
+   ```
+   aws elbv2 describe-load-balancers --query 'LoadBalancers[*].[LoadBalancerArn]' --output text
+   ```
+   This command will return a list of all the load balancer ARNs.
+
+4. Check Deletion Protection Flag: For each load balancer ARN, use the following command to check if the deletion protection flag is enabled:
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+   Replace `<load-balancer-arn>` with the ARN of the load balancer you want to check. This command will return a JSON output with the attributes of the load balancer. If the deletion protection flag is enabled, the `delete_protection.enabled` attribute will be set to `true`. If it's not, then the load balancer is misconfigured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install Boto3 in your environment. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials either by setting environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`, or by creating the credentials file at `~/.aws/credentials`.
+
+2. Import necessary libraries and create a session:
+   Import the Boto3 library in your Python script and create a session using your AWS credentials.
+   ```python
+   import boto3
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>